opentelemetry-user-events-metrics/src/tracepoint/mod.rs

```diff
@@ -67,19 +61,21 @@ pub fn enabled() -> bool {
 ///
 /// If this code is used in a shared object, the tracepoint MUST be
 /// unregistered before the shared object unloads from memory.
-pub unsafe fn register() -> i32 {
+pub unsafe fn register(trace_point: Pin<&ehi::TracepointState>) -> i32 {
     debug_assert!(METRICS_EVENT_DEF[METRICS_EVENT_DEF.len() - 1] == b'\0');
 
-    // Pin is OK because METRICS_EVENT is static.
-    // CStr::from_bytes_with_nul_unchecked is ok because METRICS_EVENT_DEF ends with "\0".
-    pin::Pin::new_unchecked(&METRICS_EVENT)
-        .register(ffi::CStr::from_bytes_with_nul_unchecked(METRICS_EVENT_DEF))
-}
-
-/// Unregisters this tracepoint.
-///
-/// Return value is 0 for success or an errno code for error. The return value is
-/// provided to help with debugging and should usually be ignored in release builds.
-pub fn unregister() -> i32 {
-    METRICS_EVENT.unregister()
+    let result = panic::catch_unwind(|| {
```

Comment:
This is pointless. Don't catch the panic. There are two cases:

- Either you create a new Tracepoint and then register it, in which case it will never panic.
- Or you have a bug in your code, and it will panic to help you find the bug.

If there is a bug in your code, you want it to panic so you can find and fix the bug. If your code is good then there will never be a panic. Instead of catching the panic (hiding the bug), you should fix your code so that you don't hit the panic.

Reply:
Removed the panic catch.